src/starbase/execution.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  writeIntegrationConfig,
+  writeNeo4jRootConfig
+} from './config';
+import { executeIntegration } from './integration';
+import { isDirectoryPresent } from '@jupiterone/integration-sdk-runtime';
+import { StarbaseConfigurationError } from './error';
+import { StarbaseConfig, StarbaseIntegration } from './types';
+
+async function setupStarbaseStorageEngine(starbaseConfig: StarbaseConfig) {
+  if (!starbaseConfig.storage) return;
+
+  if (starbaseConfig.storage.engine !== 'neo4j') {
+    throw new StarbaseConfigurationError(
+      'Invalid storage engine supplied. Neo4j is the only storage engine supported at this time.'
+    );
+  }
+
+  await writeNeo4jRootConfig(starbaseConfig.storage);
+}
+
+type OnSkipIntegrationExecutionFunctionParams = {
+  integration: StarbaseIntegration;
+  reason: string;
+};
+
+type OnExecuteStarbaseOptions = {
+  onSkipIntegrationExecution?: (
+    params: OnSkipIntegrationExecutionFunctionParams
+  ) => void
+};
+
+async function executeStarbase(
+  starbaseConfig: StarbaseConfig,
+  {
+    onSkipIntegrationExecution
+  }: OnExecuteStarbaseOptions = {}
+) {
+  await setupStarbaseStorageEngine(starbaseConfig);
+
+  for (const integration of starbaseConfig.integrations) {
+    if (!(await isDirectoryPresent(integration.directory))) {
+      if (onSkipIntegrationExecution) {
+        onSkipIntegrationExecution({
```

Comment:
We should avoid using `console.log` inside of the execution functions. I'd like `executeStarbase` to eventually collect results from each integration execution and return a summary.